typing/typemod.ml

```diff
@@ -1398,6 +1510,22 @@ and transl_signature env sg =
             mksig (Tsig_modtype mtd) env loc :: trem,
             sg :: rem,
             final_env
+        | Psig_modtypesubst pmtd ->
+            let info id tmty =
+              let mty = match tmty with
+                | Some tmty -> tmty.mty_type
+                | None -> assert false (* unparsable *)
+              in
+              let subst = Subst.add_modtype id mty Subst.identity in
+              match mty with
+              | Mty_ident _ -> `Substituted_away subst
+              | _ -> `Unpackable_modtype_substituted_away (id,subst)
+            in
+            let newenv, mtd, _sg = transl_modtype_decl ~info names env pmtd in
```

Comment:
Rather than adding an `info` parameter to `transl_modtype_decl` I think it would be cleaner and more consistent to lift the call to `Signature_names.check_modtype` out of `transl_modtype_decl` to its call sites.